haystack_experimental/components/tools/openapi/openapi_tool.py

```diff
@@ -167,6 +169,41 @@ def run(
 
         return {"service_response": response_messages}
 
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+
+        :returns:
+            The serialized component as a dictionary.
+        """
+        if "api_key" in self.generator_api_params:
+            self.generator_api_params["api_key"] = self.generator_api_params["api_key"].to_dict()
```

Comment:
This performs no validation on its assumptions. 

Looks like we don't have a `serialize_secrets_inplace` function that recursively checks values in a dict for `Secret`s and calls `to_dict` on them. Let's add that (to `haystack_experimental.util.auth`) and use that here instead.